portfolio/src/main/webapp/style.css

```diff
@@ -1,9 +1,387 @@
+}
+
+/* Image box for each project */
+.project {
+    opacity: 1;
+    display: block;
+    width: 800px;
+    height: 300px;
+    transition: .5s ease;
+    backface-visibility: hidden;
+    object-fit: cover;
+}
+
+.middle {
+    transition: .5s ease;
+    opacity: 0;
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    -ms-transform: translate(-50%, -50%);
+    text-align: center;
+}
+
+.project_container:hover .project {
+    opacity: 0.3;
+}
+
+.project_container:hover .middle {
+    opacity: 1;
+}
+
+.text {
+    background-color: #262626;
+    color: white;
+    font-size: 16px;
+    padding: 16px 32px;
+}
+
+a.image_link:link {
+    color: white;
+    background-color: transparent;
+    text-decoration: white;
+}
+
+a.image_link:visited {
+    color: white;
+    background-color: transparent;
+    text-decoration: none;
+}
+
+a.image_link:hover {
+    color: whitesmoke;
+    background-color: transparent;
+    text-decoration: underline;
 }
+
+.block {
+    display: block;
+    width: 25%;
+    border: none;
+    background-color: #c1b9f0;
+    padding: 14px 28px;
+    font-size: 16px;
+    cursor: pointer;
+    text-align: center;
+    border-radius: 12px;
+}
+
+
+/* Project slideshow */
+
+/* Slideshow container */
+.slideshow_container_vertical {
+    max-width: 400px;       /* 50% of content width */
+    position: relative;
+    margin: auto;
+}
+
+.slideshow_container {
+    max-width: 800px;
+    position: relative;
+    margin: auto;
+}
+
+/* Hide the images by default */
+.mySlides {
+    display: none;
+}
+
+/* Next and previous buttons */
+.prev, .next {
+    cursor: pointer;
+    position: absolute;
+    top: 50%;
+    width: auto;
+    margin-top: -22px;
+    padding: 16px;
+    color: white;
+    font-weight: bold;
+    font-size: 18px;
+    transition: 0.6s ease;
+    border-radius: 0 3px 3px 0;
+    user-select: none;
+}
+
+/* Position next button to the right */
+.next {
+    right: 0;
+    border-radius: 3px 0 0 3px;
+}
+
+/* On hover, add transparent black background */
+.prev:hover, .next:hover {
+    background-color: rgba(0,0,0,0.8);
+}
+
+/* Caption text */
+.slide_caption {
+    color: #000000;
+    font-size: 15px;
+    padding: 8px 12px;
+    position: absolute;
+    bottom: 8px;
+    width: 100%;
+    text-align: center;
+}
+
+/* Slideshow dots */
+.dot {
+    cursor: pointer;
+    height: 15px;
+    width: 15px;
+    margin: 0 2px;
+    background-color: #bbb;
+    border-radius: 50%;
+    display: inline-block;
+    transition: background-color 0.6s ease;
+}
+
+.active, .dot:hover {
+    background-color: #717171;
+}
+
+/* Fading animation */
+.fade {
+    -webkit-animation-name: fade;
+    -webkit-animation-duration: 1.5s;
+    animation-name: fade;
+    animation-duration: 1.5s;
+}
+
+@-webkit-keyframes fade {
+    from {opacity: .4}
+    to {opacity: 1}
+}
+
+@keyframes fade {
+    from {opacity: .4}
+    to {opacity: 1}
+}
+
+/* Gallery */
+
+.row {
```

Comment:
It seems that using &lt;div&gt;'s for layout is now preferred to &lt;table&gt;. Is that correct?

I'm trying to catch up on the current best practices advice.

Reply:
Looks like it, from the tutorials and resources I used.